### DISCUSSION TITLE
**Ubuntu 22 (Jammy Jellyfish) Truffle Install Guide**

### DISCUSSION THREAD

**Discussion by apdan510** [ConsenSys-archive/truffle#5732] [Installation]
last tested: **November 23, 2022**

The steps below have been tested on:
**Ubuntu 22.04.1 LTS (Jammy Jellyfish)** 
**Ubuntu 20.04.5 LTS (Focal Fossa)** 

Requirements:
**Node.js v14-v18 (NPM)**

**Step 1. Install Node.js via NVM**
To avoid permission errors when installing globally, Node.js (NPM) recommends installing Node.js with a [Node version manager](https://github.com/nvm-sh/nvm) (NVM) on Mac and Linux Operating systems

Install a node version manager (NVM):
Open a new terminal (you can find terminal in "Show Applications")

![image](https://user-images.githubusercontent.com/40121410/203645070-2f3cb3f2-1586-44e1-b0cc-01e91a085433.png)

Enter this command to install NVM:

`wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash`

![image](https://user-images.githubusercontent.com/40121410/203651365-d0729172-f9ea-4818-a45a-cfa60f41994b.png)

When the NVM script finishes downloading you will receive a completion message that will instruct you to close and reopen your terminal to begin using NVM. 

![image](https://user-images.githubusercontent.com/40121410/203646701-a49fc468-3d9a-41bb-a7a6-81ef27ab1b65.png)

**Step 2. Verify NVM installation**

After you close and open a new terminal, verify that NVM works by entering the command: `nvm`

![image](https://user-images.githubusercontent.com/40121410/203647314-fa9147af-1164-40fc-a864-ea757eef28f7.png)

**Step 3. Install a Truffle compatible version of Node.js**

Enter this command into your terminal: `nvm install 18`

![image](https://user-images.githubusercontent.com/40121410/203647599-5c0e3573-e318-429f-9a73-fa91bc75ee8f.png)

Verify that NPM is installed by entering: `npm`

![image](https://user-images.githubusercontent.com/40121410/203650231-5f9c2be1-8df4-4193-be93-261b486d4b01.png)

_It is good practice to run the `npm doctor` command after you set up your Node.js environment on your computer. 
[Npm doctor](https://docs.npmjs.com/cli/v7/commands/npm-doctor) checks your Npm installation if the requirements to manage your JavaScript packages are configured and installed correctly. Since we are using a compatible version of NPM to install truffle (Node 18 , Npm 8) Npm doctor may alert you that your Npm installation to needs an update. This is because Npm doctor always looks at the latest version of Npm version available. You can ignore this warning and continue onto installing Truffle in the next step._ 

![image](https://user-images.githubusercontent.com/40121410/203654079-4bce190f-65ae-454a-9f89-0e6fbc8f20d2.png)

**Step 4. Install Truffle**
Enter this command in a new terminal: `npm install -g truffle`

![image](https://user-images.githubusercontent.com/40121410/203648268-2f3cc566-dc5a-461b-afaf-4da6e9089614.png)

You may see some warnings or conflicts, this is typical and should work normally as long as there are no errors. 

**Step 5. Verify that Truffle is installed**

Enter this command to verify the Truffle installation:` truffle version`

![image](https://user-images.githubusercontent.com/40121410/203648475-451275ad-1307-4e4a-bbbf-25535fe51c33.png)

**Comment by 12345662312** [comment]
Thanks a lot
